src/candid_rpc.rs

```diff
@@ -61,13 +58,23 @@ impl RpcTransport for CanisterTransport {
     async fn http_request(
         _provider: &RpcService,
         request: CanisterHttpRequestArgument,
-        cost: u128,
-    ) -> CallResult<HttpResponse> {
-        Ok(
-            ic_cdk::api::management_canister::http_request::http_request(request, cost)
-                .await?
-                .0,
-        )
+        cycles: u128,
+    ) -> RpcResult<HttpResponse> {
+        if !is_authorized(&ic_cdk::caller(), Auth::FreeRpc) {
+            let cycles_available = ic_cdk::api::call::msg_cycles_available128();
+            if cycles_available < cycles {
+                return Err(ProviderError::TooFewCycles {
+                    expected: cycles,
+                    received: cycles_available,
+                }
+                .into());
+            }
+            ic_cdk::api::call::msg_cycles_accept128(cycles);
```

Comment:
I'm a bit confused about the logic here: Shouldn't the attached cycles match a predetermined amount required to make the outcall? What is the point of passing along a cycles parameter that may or may not be sufficient (nor necessarily the same as the number of attached cycles)?

Why is any amount greater than `cycles` allowed?

Reply:
`cycles` (now renamed to `cycles_cost` for clarity) is the cost calculated by the ckETH canister based on the HTTPS outcall. `msg_cycles_accept128()` only consumes this cost, so you can send more cycles than necessary without being overcharged.